logins-api/android/library/src/main/java/org/mozilla/sync15/logins/LoginsStorage.kt

```diff
@@ -59,4 +59,35 @@ interface LoginsStorage : Closeable {
      * Fetch the full list of passwords from the underlying storage layer.
      */
     fun list(): SyncResult<List<ServerPassword>>
+
+    /**
+     * Insert the provided login into the database.
+     *
+     * This function ignores values in metadata fields (`timesUsed`,
+     * `timeCreated`, `timeLastUsed`, and `timePasswordChanged`).
+     *
+     * A GUID may be provided, however if `login` has an empty
```

Comment:
uber-nit: This is just a little awkward.  Maybe just: "If `login` does not have a `id` field, or has an empty `id` field, then a GUID will be generated automatically."  What style of GUID do we generate?  (8 chars or "{guid}" style?)

Reply:
12 base64url encoded random characters.